### PR TITLE
Undo elf fix

### LINKED ISSUE #2725: Revert hacky change mitigating elf overeaching thread limiting
once https://github.com/constantinpape/elf/issues/75 is merged and we've updated our elf dependency, we can revert https://github.com/ilastik/ilastik/commit/921efda35ee718f9644e76477e20955dfe93e30f

### PR DESCRIPTION
**Pull request by k-dominik**
our hacky fix is not necessary when ensuring elf > 0.4.8

Summary:
* updated pin for `elf`
* updated dependencies for pandas2 (elf --> skan2 --> pandas2)
* removed the hacky fix from #2724
* plus only very vaguely related changes
  * watershed in muticut is sensitive to small numerical inaccuracies in filters. The superpixels are not the same on OSX arm-64. In order to still test the RF training there, added a test that uses ground truth to autolabel. Related to that, we require a slightly lower accuracy (0.98 -> 0.97) to compare superpixels to ground truth.
  * Moved some of the testing utility functions into a single module (`tests.test_ilastik.helpers.wait`) in order to reuse them in the multicut test. There is some overlap in the functions there, but will not solve this now.
  * re-enabled Carving tests on osx - they work now (with python>=3.8) 

xref: #2724
fixes #2725

### CODE REVIEW

File: tests/test_ilastik/test_workflows/testCarvingGui.py [hunk 1]
```diff
@@ -34,14 +34,15 @@
 
 
 # https://bugreports.qt.io/browse/QTBUG-87014
-is_darwin_bigsur = False
+must_skip_darwing_bigsur = False
 if platform.system().lower() == "darwin":
     mac_ver = tuple(map(int, platform.mac_ver()[0].split(".")))
-    if mac_ver >= (10, 6):
-        is_darwin_bigsur = True
+    python_ver = tuple(map(int, platform.python_version().split(".")))
```
Comment (emilmelnikov) [inline tests/test_ilastik/test_workflows/testCarvingGui.py:40]: I wonder if it's time to move this into some `version` utility module.

File: tests/test_ilastik/helpers/wait.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+def wait_until(f, timeout=10):
+    result = f()
+    while not result and timeout:
+        timeout -= 1
+        time.sleep(1)
+        result = f()
+    if not result:
+        raise Exception(f"TIMEOUT! waiting for {f}")
+    return result
```
Comment (emilmelnikov) [inline tests/test_ilastik/helpers/wait.py:38]:
```suggestion
def wait_until(f, seconds=10):
    for t in range(seconds + 1):
        time.sleep(t > 0)
        if result := f():
            return result
    raise Exception(f"TIMEOUT! waiting for {f}")
```

File: tests/test_ilastik/helpers/wait.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    :param timeout: timeout in ms
+    :param signal:
+    """
+    evtLoop = QEventLoop()
+    # QueuedConnection so we don't receive signal before entering the loop
+    signal.connect(evtLoop.quit, Qt.QueuedConnection)
+
+    yield
+
+    _exec_with_timeout(evtLoop, timeout, signal)
```
Comment (emilmelnikov) [inline tests/test_ilastik/helpers/wait.py:55]: Wrap `yield` in `try...finally`?

**Review by emilmelnikov — APPROVED** [review]
🧝‍♂️🧝‍♀️

File: tests/test_ilastik/helpers/wait.py (hunk 2 above)
Comment (k-dominik) [reply tests/test_ilastik/helpers/wait.py:38]: :walrus:

File: tests/test_ilastik/helpers/wait.py (hunk 3 above)
Comment (k-dominik) [reply tests/test_ilastik/helpers/wait.py:55]: yeah will be nicer to handle quitting of loop in this function

**Review by btbest — APPROVED** [review]
Agree with Emil's comments and nothing further 🆒

File: tests/test_ilastik/test_workflows/testCarvingGui.py [hunk 4]
```diff
@@ -25,24 +25,15 @@
 logger.addHandler(logging.StreamHandler(sys.stdout))
 logger.setLevel(logging.DEBUG)
 
-import platform
 import pytest
 
 
 if os.environ.get("ON_CIRCLE_CI"):
     pytest.skip("These tests are too flaky on circleCI", allow_module_level=True)
+elif not is_mac_py_compatible():
```
Comment (emilmelnikov) [inline tests/test_ilastik/test_workflows/testCarvingGui.py:33]: Hmm, now looking at this I wonder whether it'd be better to check for `os.environ.get("VOLUMINA_SHOW_3D_VIDGET", "1") == "0"`...

File: tests/test_ilastik/test_workflows/testCarvingGui.py (hunk 1 above)
Comment (k-dominik) [reply tests/test_ilastik/test_workflows/testCarvingGui.py:40]: this change introduced some side effects, so with @emilmelnikov, we decided to leave it as is...

**Pull request merged**